### PR TITLE
Feature/update logs

### PR DESCRIPTION
**Pull request by SamnitPatil9882**
debug logs added
log level change handler added

### CODE REVIEW

File: internal/pkg/logger/log.go [hunk 1]
```diff
@@ -1,25 +1,108 @@
...
+	Compress:   true,
+}
+
 // Error - prints out an error
-func Error(appError error, msg string, triggeringError error) {
-	l.WithFields(l.Fields{"appError": appError, "message": msg}).Error(triggeringError)
+func Error(ctx context.Context, args ...interface{}) {
+	requestID, ok := ctx.Value(constants.RequestID).(string)
+	if !ok {
+		requestID = ""
```
Comment (yama11299) [inline internal/pkg/logger/log.go:28]: instead of empty use "N/A"

File: internal/pkg/middleware/jwtMiddleware.go [hunk 2]
```diff
@@ -60,9 +61,20 @@ func JwtAuthMiddleware(next http.Handler, roles []string) http.Handler {
...
 
 		next.ServeHTTP(rw, req)
 
 	})
 }
+
+// RequestIDMiddleware generates a request ID and adds it to the request context.
+func RequestIDMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		reqestId := uuid.NewString()
```
Comment (yama11299) [inline internal/pkg/middleware/jwtMiddleware.go:75]: fix requestID typo

File: internal/repository/postgresdb/appreciation.go [hunk 3]
```diff
@@ -112,33 +116,37 @@ func (appr *appreciationsStore) GetAppreciationById(ctx context.Context, tx repo
 		ToSql()
 
 	if err != nil {
-		logger.Errorf("error in generating squirrel query, err: %v", err)
+		logger.Errorf(ctx,"error in generating squirrel query, err: %v", err)
 		return repository.AppreciationResponse{}, apperrors.InternalServer
 	}
 
+	logger.Debug(ctx," appr: query: ",query)
+	logger.Debug(ctx," appr: args: ",args)
```
Comment (yama11299) [inline internal/repository/postgresdb/appreciation.go:124]: don't log query and parameter separately.

File: internal/api/router.go [hunk 4]
```diff
@@ -22,7 +22,13 @@ func NewRouter(deps app.Dependencies) *mux.Router {
 	router := mux.NewRouter()
 	peerlySubrouter := router.PathPrefix("/peerly").Subrouter()
 
+	// Add the RequestIDMiddleware to the subrouter
+	peerlySubrouter.Use(middleware.RequestIDMiddleware)
+
 	peerlySubrouter.HandleFunc("/ping", pingHandler).Methods(http.MethodGet)
+
+	peerlySubrouter.HandleFunc("/set_logger_level", loggerHandler).Methods(http.MethodPost)
```
Comment (yama11299) [inline internal/api/router.go:30]: method PATCH

File: internal/api/appreciation.go [hunk 5]
```diff
@@ -9,62 +9,68 @@ import (
...
-	logger "github.com/sirupsen/logrus"
 )
 
 func createAppreciationHandler(appreciationSvc appreciation.Service) http.HandlerFunc {
 	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
 		var appreciation dto.Appreciation
 		err := json.NewDecoder(req.Body).Decode(&appreciation)
 		if err != nil {
-			logger.Errorf("Error while decoding request data : %v", err)
+			log.Errorf(req.Context(),"Error while decoding request data : %v", err)
```
Comment (yama11299) [inline internal/api/appreciation.go:21]: why request.Context() everywhere?

File: internal/api/appreciation.go [hunk 6]
```diff
@@ -9,62 +9,68 @@ import (
...
+		log.Info(req.Context(),"getAppreciationByIDHandler: request: ",req)
 		resp, err := appreciationSvc.GetAppreciationById(req.Context(), int32(apprID))
 		if err != nil {
-			logger.Errorf("err : %v", err)
+			log.Errorf(req.Context(),"getAppreciationByIDHandler: err : %v", err)
 			dto.ErrorRepsonse(rw, err)
 			return
 		}
+
+		log.Debug(req.Context(),"getAppreciationByIDHandler: request: ",resp)
```
Comment (yama11299) [inline internal/api/appreciation.go:65]: fix the typo error and check all logs for the same.

File: internal/api/appreciation.go [hunk 7]
```diff
@@ -9,62 +9,68 @@ import (
...
+		log.Info(req.Context(),"Appreciation data got successfully")
 		dto.SuccessRepsonse(rw, http.StatusOK, "Appreciation data got successfully", resp)
```
Comment (yama11299) [inline internal/api/appreciation.go:67]: update the result statement.

File: internal/pkg/logger/log.go (hunk 1 above)
Comment (SamnitPatil9882) [reply internal/pkg/logger/log.go]: created new function and if requestID is empty "N/A" used

File: internal/pkg/middleware/jwtMiddleware.go (hunk 2 above)
Comment (SamnitPatil9882) [reply internal/pkg/middleware/jwtMiddleware.go]: fixed

File: internal/repository/postgresdb/appreciation.go (hunk 3 above)
Comment (SamnitPatil9882) [reply internal/repository/postgresdb/appreciation.go]: resolved

File: internal/api/router.go (hunk 4 above)
Comment (SamnitPatil9882) [reply internal/api/router.go]: resolved

File: internal/api/appreciation.go (hunk 5 above)
Comment (SamnitPatil9882) [reply internal/api/appreciation.go:21]: created variable
ctx := req.Context()
and used it

File: internal/api/appreciation.go (hunk 6 above)
Comment (SamnitPatil9882) [reply internal/api/appreciation.go:65]: resolved

File: internal/api/appreciation.go (hunk 7 above)
Comment (SamnitPatil9882) [reply internal/api/appreciation.go]: changed to "Appreciation data fetched successfully"

File: internal/api/appreciation.go [hunk 8]
```diff
@@ -9,64 +9,72 @@ import (
...
 			err = apperrors.JSONParsingErrorReq
 			dto.ErrorRepsonse(rw, err)
 			return
 		}
 
+		log.Debug(ctx, "createAppreciationHandler: request: ", req)
 		err = appreciation.ValidateCreateAppreciation()
 		if err != nil {
-			logger.Errorf("Error while validating request data : %v", err)
+			log.Errorf(req.Context(), "Error while validating request data : %v", err)
```
Comment (yama11299) [inline internal/api/appreciation.go:31]: again req.Context()

Comment (SamnitPatil9882) [reply internal/api/appreciation.go]: apologize
removed all req.Context() from logs

**Comment by yama11299** [comment]
@SamnitPatil9882 Please :pray: tell me why don't you "go fmt" the code, I have already asked you several time to "go fmt", "go lint", "go vet" the code.

cc: @shailendrakanherkar18

File: internal/api/appreciation.go [hunk 9]
```diff
@@ -76,33 +84,39 @@ func listAppreciationsHandler(appreciationSvc appreciation.Service) http.Handler
...
-		err = appreciationSvc.DeleteAppreciation(req.Context(), int32(apprId))
+		log.Info(ctx, "deleteAppreciationHandler: request: ", req)
+		err = appreciationSvc.DeleteAppreciation(ctx, int32(apprId))
 		if err != nil {
-			logger.Errorf("err : %v", err)
+			log.Errorf(ctx, "deleteAppreciationHandler: err : %v", err)
 			dto.ErrorRepsonse(rw, err)
 			return
 		}
+		log.Debug(ctx, "deleteAppreciationHandler: resp: ", err)
```
Comment (yama11299) [inline internal/api/appreciation.go:118]: err?

File: internal/app/appreciation/service.go [hunk 10]
```diff
@@ -42,75 +43,81 @@ func (apprSvc *service) CreateAppreciation(ctx context.Context, appreciation dto
...
+	logger.Debug(ctx, "service: CreateAppreciation: appreciation: ", appreciation)
 
 	//add sender
 	data := ctx.Value(constants.UserId)
 	sender, ok := data.(int64)
 	if !ok {
-		logger.Error("err in parsing userid from token")
+		logger.Error(ctx, "service: err in parsing userid from token")
 		return dto.Appreciation{}, apperrors.InternalServer
 	}
```
Comment (yama11299) [inline internal/app/appreciation/service.go:54]: should create a method in contextstore and use it

File: internal/app/appreciation/service.go [hunk 11]
```diff
@@ -135,74 +142,81 @@ func (apprSvc *service) CreateAppreciation(ctx context.Context, appreciation dto
...
 
 	userBadgeDetails, err := apprSvc.appreciationRepo.UpdateUserBadgesBasedOnTotalRewards(ctx, tx)
 	if err != nil {
-		logger.Error("err: ", err.Error())
+		logger.Error(ctx, "service: err: ", err.Error())
 		return false, err
 	}
-
+	logger.Debug(ctx, "service: UpdateAppreciation: ", userBadgeDetails)
 	apprSvc.sendEmailForBadgeAllocation(userBadgeDetails)
```
Comment (yama11299) [inline internal/app/appreciation/service.go:220]: pass ctx to sendEmail methods

File: internal/app/reward/service.go [hunk 12]
```diff
@@ -32,123 +32,134 @@ func NewService(rewardRepo repository.RewardStorer, appreciationRepo repository.
...
 		return dto.Reward{}, apperrors.InternalServer
 	}
 	rewardReq.SenderId = sender
 
 	appr, err := rwrdSvc.appreciationRepo.GetAppreciationById(ctx, nil, int32(rewardReq.AppreciationId))
 	if err != nil {
+		logger.Errorf(ctx, "rwrdSvc: gerAppreciationById err : %v", err)
 		return dto.Reward{}, err
 	}
+	logger.Debug(ctx, "rwrdSvc: appr: ", appr)
```
Comment (yama11299) [inline internal/app/reward/service.go:50]: use full name -> reward service

File: internal/app/reward/service.go [hunk 13]
```diff
@@ -32,123 +32,134 @@ func NewService(rewardRepo repository.RewardStorer, appreciationRepo repository.
...
 	//deduce user rewardquota
 
 	deduceChk, err := rwrdSvc.rewardRepo.DeduceRewardQuotaOfUser(ctx, tx, rewardReq.SenderId, int(rewardReq.Point))
 	if err != nil {
+		logger.Errorf(ctx, "rwrdSvc: DeduceRewardQuotaOfUser: err: %v", err)
 		return dto.Reward{}, err
 	}
 
 	if !deduceChk {
+		logger.Error(ctx, "rwrdSvc: DeduceRewardQuotaOfUser")
```
Comment (yama11299) [inline internal/app/reward/service.go:123]: What do you understand by this log?

File: internal/app/reward/service.go [hunk 14]
```diff
@@ -32,123 +32,134 @@ func NewService(rewardRepo repository.RewardStorer, appreciationRepo repository.
...
-
 func (rwrdSvc *service) sendRewardNotificationToSender(ctx context.Context, user dto.GetUserByIdResp) {
 
+	logger.Debug(ctx, " rwrdSvc: sendRewardNotificationToSender: user: ", user)
 	notificationTokens, err := rwrdSvc.userRepo.ListDeviceTokensByUserID(ctx, user.UserId)
+	logger.Debug(ctx, " notificationTokens: ", notificationTokens)
 	if err != nil {
-		logger.Errorf("err in getting device tokens: %v", err)
-		return 
+		logger.Errorf(ctx, "err in gettinsendRewardNotificationToSenderg device tokens: %v", err)
```
Comment (yama11299) [inline internal/app/reward/service.go:153]: fix typo

File: internal/app/reward/service.go [hunk 15]
```diff
@@ -32,123 +32,134 @@ func NewService(rewardRepo repository.RewardStorer, appreciationRepo repository.
...
+		return
 	}
 
 	msg := notification.Message{
 		Title: "Reward Given Successfully",
 		Body:  "You have successfully given a reward! ",
 	}
 
+	logger.Debug(ctx, " msg: ", msg)
```
Comment (yama11299) [inline internal/app/reward/service.go:162]: no need to log the hardcoded data.

File: internal/app/users/service.go [hunk 16]
```diff
@@ -95,30 +95,30 @@ func (us *service) GetIntranetUserData(ctx context.Context, req dto.GetIntranetU
...
 		return
 	}
 
 	intranetReq.Header.Add(constants.AuthorizationHeader, req.Token)
 	resp, err := client.Do(intranetReq)
 	if err != nil {
-		logger.Errorf("error in intranet get user api. status returned: %d, err: %s  ", resp.StatusCode, err.Error())
-		logger.Errorf("error response: %v", resp)
+		logger.Errorf(ctx, "error in intranet get user api. status returned: %d, err: %s  ", resp.StatusCode, err.Error())
+		logger.Errorf(ctx, "error response: %v", resp)
```
Comment (yama11299) [inline internal/app/users/service.go:107]: can't this be logged with above log statement?

File: internal/app/appreciation/service.go (hunk 11 above)
Comment (SamnitPatil9882) [reply internal/app/appreciation/service.go:220]: Change is present in Next PR

File: internal/pkg/logger/log.go [hunk 17]
```diff
@@ -1,25 +1,105 @@
...
+	})
+
+	// Set the logging level
+	logger.SetLevel(l.InfoLevel)
+
+	Logger = logger
+	return logger, nil
+}
+
+func setRequestId(ctx context.Context) *l.Entry {
```
Comment (yama11299) [inline internal/pkg/logger/log.go:89]: rename this function or create separate function to set userID in logger

File: internal/pkg/middleware/jwtMiddleware.go [hunk 18]
```diff
@@ -59,9 +60,20 @@ func JwtAuthMiddleware(next http.Handler, role int) http.Handler {
...
 		next.ServeHTTP(rw, req)
 
 	})
 }
+
+// RequestIDMiddleware generates a request ID and adds it to the request context.
+func RequestIDMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		requestID := uuid.NewString()
+		ctx := context.WithValue(r.Context(), constants.RequestID, requestID)
```
Comment (yama11299) [inline internal/pkg/middleware/jwtMiddleware.go:75]: set request ID only if not present in request already

File: internal/repository/postgresdb/appreciation.go [hunk 19]
```diff
@@ -223,51 +235,56 @@ func (appr *appreciationsStore) ListAppreciations(ctx context.Context, tx reposi
...
-	fmt.Println("id -> ", id)
+	logger.Debug(ctx, " id -> ", id)
 	userId, ok := ctx.Value(constants.UserId).(int64)
 	if !ok {
-		logger.Error("unable to convert context user id to int64")
+		logger.Error(ctx, "unable to convert context user id to int64")
 		return nil, repository.Pagination{}, apperrors.InternalServerError
 	}
 
+	logger.Debug(ctx, " userId: ", userId)
```
Comment (yama11299) [inline internal/repository/postgresdb/appreciation.go:263]: repeated code

File: internal/repository/postgresdb/rewards.go [hunk 20]
```diff
@@ -25,6 +25,7 @@ func NewRewardRepo(db *sqlx.DB) repository.RewardStorer {
 
 func (rwrd *rewardStore) GiveReward(ctx context.Context, tx repository.Transaction, reward dto.Reward) (repository.Reward, error) {
 
+	logger.Debug(ctx, fmt.Sprintf("rwrdRepo: GiveReward: %v", reward))
```
Comment (yama11299) [inline internal/repository/postgresdb/rewards.go:28]: is it needed to use fmt.Printf()

File: internal/api/appreciation.go (hunk 9 above)
Comment (SamnitPatil9882) [reply internal/api/appreciation.go:118]: removed

File: internal/app/appreciation/service.go (hunk 10 above)
Comment (SamnitPatil9882) [reply internal/app/appreciation/service.go:54]: will create contextstore and methods in seperate PR

File: internal/app/reward/service.go (hunk 13 above)
Comment (SamnitPatil9882) [reply internal/app/reward/service.go:123]: error message updated

File: internal/app/reward/service.go (hunk 14 above)
Comment (SamnitPatil9882) [reply internal/app/reward/service.go:153]: updated

File: internal/app/reward/service.go (hunk 15 above)
Comment (SamnitPatil9882) [reply internal/app/reward/service.go:162]: line removed

File: internal/app/users/service.go (hunk 16 above)
Comment (SamnitPatil9882) [reply internal/app/users/service.go:107]: updated

File: internal/pkg/logger/log.go (hunk 17 above)
Comment (SamnitPatil9882) [reply internal/pkg/logger/log.go:89]: updated to func getLoggerWithRequestContext

File: internal/pkg/middleware/jwtMiddleware.go (hunk 18 above)
Comment (SamnitPatil9882) [reply internal/pkg/middleware/jwtMiddleware.go:75]: Peerly routes are not called from other services; therefore, the request ID will not be present.

File: internal/repository/postgresdb/rewards.go (hunk 20 above)
Comment (SamnitPatil9882) [reply internal/repository/postgresdb/rewards.go:28]: updated

**Comment by yama11299** [comment]
approved.

**Pull request merged**